### ISSUE TITLE
Can not imported ScreenBoard got an error unmarshal

### ISSUE THREAD

**Issue by officel** [terraform-providers/terraform-provider-datadog#105]
Hi there,

I'm happy to be able to manage the screenboard with DataDog.
But I tried importing an existing board but it got an error.
Am I something wrong?

### Terraform Version

```
$ terraform -v
Terraform v0.11.8
+ provider.aws v1.40.0
+ provider.datadog v1.4.0
```

### Affected Resource(s)
- datadog_screenboard

### Terraform Configuration Files
```hcl
resource "datadog_screenboard" "security_log" {
}
```
Of course, other resources can be managed.

### Expected Behavior

imported screenboard

### Actual Behavior

```
$ terraform import datadog_screenboard.security_log 465952
datadog_screenboard.security_log: Importing from ID "465952"...

Error: datadog_screenboard.security_log (import id: 465952): import datadog_screenboard.security_log (id: 465952): json: cannot unmarshal number into Go struct field Screenboard.height of type string
```

### Steps to Reproduce

only use import .

### Important Factoids

maybe any boards cannot import...

aws integration make some boards, can not imported.

### References

https://www.terraform.io/docs/providers/datadog/r/screenboard.html

Please tell me if there is any information. Thank you.

**Comment by tobypinder** [comment]
100% experiencing the same issue (can provide screenboard id on request).

**Comment by guitarrapc** [comment]
same error on my account, too.

> son: cannot unmarshal number into Go struct field Screenboard.height of type string

**Comment by masci** [comment]
This is related to an issue we're tracking with #117, let's use that to consolidate communications.

**Comment by rodush** [comment]
Same here, tried plugin v1.5.0 and 1.6.0; `terraform -v: 0.11.11`:
> * module.screenboards.datadog_screenboard.help_screenboard: datadog_screenboard.help_screenboard: json: cannot unmarshal number into Go struct field Screenboard.height of type string

ℹ️  `help_screenboard` is the resource name in our specific case, not related to the issue directly.

**Comment by rodush** [comment]
@masci would you mind reopening this? Exactly the same issue!
Our assumption is it looks like if one modifies the Dashboard in Datadog manually, then it can not pick up by terraform/datadog-provider.

**Comment by rodush** [comment]
I wonder, can it be source of problem? https://github.com/terraform-providers/terraform-provider-datadog/blob/master/datadog/resource_datadog_screenboard.go#L555

**Comment by rodush** [comment]
Hi! @masci did you have time to look into it, by any chance?

**Comment by bondezbond** [comment]
any update on this error? I'm having the same issue and wanted to manage the boards via terraform

**Comment by officel** [comment]
Errors will continue to occur with provider.datadog v1.7.0 ...

```
terraform import datadog_screenboard.context 421493
datadog_screenboard.context: Importing from ID "421493"...

Error: datadog_screenboard.context (import id: 421493): import datadog_screenboard.context (id: 421493): json: cannot unmarshal string into Go struct field Screenboard.width of type int
```

**Comment by officel** [comment]
Organize the plan additionally

```
$ terraform plan
  ~ datadog_screenboard.context
      height:               "768" => ""
      widget.0.height:      "20" => "15"
      widget.0.text_size:   "" => "auto"
      widget.0.title_align: "" => "left"
      widget.0.title_size:  "0" => "16"
      widget.0.url:         "https://path/to/image.jpg" => ""
      widget.0.width:       "32" => "50"
      widget.0.x:           "32" => "0"
      widget.0.y:           "7" => "0"
      width:                "1024" => ""
```

width and height is auto fix on DataDog...?

**Comment by bondezbond** [comment]
i think this might have to do with higher issue in the library upstream. After looking through the code. Not a go expert though. https://github.com/zorkian/go-datadog-api/pull/219

**Comment by mohnishbasha** [comment]
I am also facing the same issue importing a screen board.
`Error: datadog_screenboard.dashboard (import id: 276895): import datadog_screenboard.dashboard (id: 276895): json: cannot unmarshal string into Go struct field Screenboard.width of type int
`

**Comment by wbuiejr** [comment]
I also saw an issue with not being able to import a screenboard that had been touched by human hands. Regrettably, WYSIWYG is the easiest way to develop a screenboard.

When I was working with this, I noticed there was a 'palette_flip' flip attribute on a host map widget which was defined with a boolean value of 'true' in my terraform code, which functions correctly when applied to Datadog, whose value was represented in the JSON (which I could see when editing the screenboard) as the numeral '1'.  This was causing an error in Datadog when I was trying to modify a clone of this screenboard.  When I edited the JSON backend to spell out "true", the screenboard was happy with it.

I offer this clue in hope that it might help a source maintainer identify where the problem lies.

**Comment by facundovictor** [comment]
I'm facing the same problem and it is preventing me to update the provider even to 1.5:
```
json: cannot unmarshal string into Go struct field Screenboard.height of type int
```
This will be an issue to go to terraform 0.12.X soon, as I can't update the provider without removing the the dashboard from the state.

**Comment by bkabrda** [comment]
Hey folks 👋 due to some changes in the backend code, this shouldn't be happening as of August 20th 2019. Can you please try again and confirm that it's the case? Thanks!

**Comment by officel** [comment]
@bkabrda 

Thanks for the great news.
But unfortunately the problem has not been solved yet.
It looks like another problem.

can you check this sample?
https://www.terraform.io/docs/providers/datadog/r/screenboard.html

```
resource "datadog_screenboard" "test" {
  title  = "Test Screenboard"
  # width  = 0
  # height = 0
  widget {
    type       = "free_text"
    x          = 5
    y          = 5
    text       = "test text"
    text_align = "right"
    font_size  = "36"
    color      = "#ffc0cb"
  }

}
```

- width and height should be optional but always seem to be updated to null.
- set width and height to 0, it seems that it will not be changed after that
- and dashboard is generated, but no widget seems to be created.

**Comment by officel** [comment]
Widgets are created after several tests of apply & destroy...

**Comment by bkabrda** [comment]
Yeah, I can reproduce the issue with non-empty TF plan when width and height are not set. I'll submit a PR to get that fixed. The widget seems to be created for me (and it was for you eventually as well, so I'm going to assume that this is working fine).

**Comment by bkabrda** [comment]
So I closed this by #404 - please give this a try when the new version of the provider gets released and open a new issue if there are other errors.

Also consider trying the new `datadog_dashboard` resource, which unifies screenboards and timeboards and we recommend using that instead of the old `datadog_screenboard` and `datadog_timeboard` resources: https://www.terraform.io/docs/providers/datadog/r/dashboard.html
